### ISSUE TITLE
Require version number in JSON schema 

### ISSUE THREAD

**Issue by ibarra-michelle** [US-EPA-CAMD/PI-1#6339]
## Description
When utilizing a json file that does not include a version data element (e.g.  "version": "1.0.0",), then going through a series of steps adding, importing, deleting, and importing the file again, ECMPS 2.0 will still display the deleted component data in the UI, even though the latest imported file does not include component data. When JSON files with the version number that have componentID data deleted and then the file is imported, ECMPS 2.0 does not appear to display the delete componentID data (i.e., UI display matches the file data as expected).
![Image](https://github.com/user-attachments/assets/7e8c51ec-faa2-4f09-8afc-e7fc775c25da)

## Steps to reproduce the issue
1. Login to ECMPS as a data submitter, navigate to the MP module, Open and Checkout the following config: ORISE 470 Location 1. 
2. Import a file where new componentID data has been added and the previous utilized componentID now has an endDate and endHour. I added an endDate and endHour to componentID 113 and added new componentID 130 See file-1....json.
3. Navigate to the Sections dropdown, select Systems, select System ID 112, and you can see the new component data for componentIDs 113 and 130.
4. Modify the file to delete the componentID 130 data and remove the an endDate and endHour of componentID 113. See file-2....json.
5. Import file-2....json into ECMPS.
6. Navigate to the Sections dropdown, select Systems, select component 112, and you can still see the componentID 130 data that was deleted the file yet still in displayed in ECMPS. 
![Image](https://github.com/user-attachments/assets/fde5c266-a801-4bdb-9b58-4ebfae7cc0d2)

[file-2-MP Export - Comanche (470), undefined (6-24-2024).json](https://github.com/user-attachments/files/16474769/file-2-MP.Export.-.Comanche.470.undefined.6-24-2024.json)
[file-1-MP Export - Comanche (470), undefined (6-24-2024).json](https://github.com/user-attachments/files/16474770/file-1-MP.Export.-.Comanche.470.undefined.6-24-2024.json)

**Acceptance Criteria**
* Test import of JSON file for all data types
* Test Import from historical for all data types

**Comment by annalbrecht** [comment]
Document the requirements and then implement them.

**Comment by yonatan-dp** [comment]
### Acceptance Criteria for JSON Schema Import

**1. Version Attribute Requirement**
   - **Description**: No JSON Schema should be imported without a version attribute.
   - **Acceptance Criteria**:
     - **Criterion 1**: If a JSON Schema does not contain a version attribute, it should be rejected.
     - **Criterion 2**: The rejection should be accompanied by an error message: "Error: JSON Schema is missing the required version attribute."

**Comment by ibarra-michelle** [comment]
> ### Acceptance Criteria for JSON Schema Import
@yonatan-dp should an AC to update the "lastUpdated" schema field be added to the list of ACs you provided? Thx

@ibarra-michelle, I have updated the acceptance criteria to include that.

**Comment by maxdiebold-erg** [comment]
In addition to adding this version check to the Monitoring Plan import process, the check should be added to the import processes of the **Emissions** and **QA** modules.

**Comment by mark-hayward-erg** [comment]
@usmanahmederg @maxdiebold-erg please revert the merge of the associated PRs from the develop branches. We are not yet ready to merge Sprint 11 PRs into develop.

**Comment by mark-hayward-erg** [comment]
@usmanahmederg you will need to recreate the original PRs so the necessary changes can be merged into develop at the appropriate time.

**Comment by lgiannini1** [comment]
Not sure if this issue is related to this ticket, but the following errors are returned when trying to import QA data from historical. This error was returned for ORIS 147, K-7 and ORIS 3, 1, 2, CS0AAN for all quarters I tried (2023 Q1 through 2023 Q3)

![Image](https://github.com/user-attachments/assets/8989a290-b59e-4775-a45f-cfc8bad90ca5)

**Comment by mark-hayward-erg** [comment]
> Not sure if this issue is related to this ticket, but the following errors are returned when trying to import QA data from historical. This error was returned for ORIS 147, K-7 and ORIS 3, 1, 2, CS0AAN for all quarters I tried (2023 Q1 through 2023 Q3)
> 
> ![Image](https://github.com/user-attachments/assets/8989a290-b59e-4775-a45f-cfc8bad90ca5)

Yes.   These errors indicate that the logic used to generate the JSON from historical also needs to be updated.

**Comment by mxtomoto1** [comment]
Files (QA test, emissions) are able to be imported from historical successfully . All exported JSONs show version number, and files (MP, QA test, cert event, TEE, emissions) with version number omitted cannot be imported.

<img width="705" alt="t_d1" src="https://github.com/user-attachments/assets/bb9120df-be2e-48aa-a281-d935a614715b">

(As a side note, only MP shows one error message, whereas the other modules also state, "Version number must be a string.")

**Comment by ibarra-michelle** [comment]
Verified in test for various facilities (ORIS 470 Config 2; Silverhawk 55841 Config A01; ORIS 612 Config FMCT2A):

MP
- Removed version
![Image](https://github.com/user-attachments/assets/37f5cf3c-8471-48f1-a002-d7388c5cef50)
- Validated version
- ![Image](https://github.com/user-attachments/assets/54941cdc-e47f-49a0-b422-b89bb1332a0d)

QA
QA - Test Data
-Removed version
![Image](https://github.com/user-attachments/assets/06cce458-d19a-4ec2-a38b-770b5b12da59)
- Validated version
- ![Image](https://github.com/user-attachments/assets/85b5c368-5229-43ae-a871-735bbd70a464)


QA- Cert Event
- Removed version
- ![Image](https://github.com/user-attachments/assets/11c809ab-f2a4-4118-9cab-4c2e114ed3f5)
- Validated version
- ![Image](https://github.com/user-attachments/assets/599caae0-2213-460c-82b2-a69e9bcea569)


QA -TEE
- Removed version
- ![Image](https://github.com/user-attachments/assets/047c1b2a-33d1-49f7-9488-f025f4afca82)
- Validated version
- ![Image](https://github.com/user-attachments/assets/9ff0fa3e-9be1-48e7-91ee-117c5b89dfce)



Emissions
- Removed version
- ![Image](https://github.com/user-attachments/assets/810d40ac-5bba-42fc-8b8c-0349c4d21fff)
- Validated version
![Image](https://github.com/user-attachments/assets/ecf0fe33-7f73-45a5-bf6d-6edbf9f205e9)
